### PR TITLE
misc(ticdc): add more linters and fix issues found by misspell

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve? 

- Fix misspelled words
- Add more linters to avoid future problems (by copying from the pingcap/tidb `.golangci.yml`)

Differences with pingcap/tidb
- G404 disabled for `gosec` (`G404: Use of weak random number generator (math/rand instead of crypto/rand)`, should be fine as the uses are in non-crypto contexts, like retries)
- `bodyclose` removed
- `rowserrcheck` removed
- `unconvert` removed

Would be great to enable the removed checks once the issues found by them are fixed. Annotations for G404 could be used (`// #nosec G404`) if we want to enable this check.

### Check List 

Tests 

 - No code

### Release note 

```release-note
None
```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -3,6 +3,22 @@ linters:
     - unconvert
     - unparam
     - revive
+    - misspell
```
Comment (Ehco1996) [inline .golangci.yml:6]: could you please also enable these checker for dm? https://github.com/pingcap/ticdc/blob/master/dm/.golangci.yml

**Review by overvenus — APPROVED** [review]
LGTM, thanks!

File: .golangci.yml (hunk 1 above)
Comment (dveeden) [reply .golangci.yml:6]: It's already included: https://github.com/pingcap/ticdc/blob/71e2a4818e91b8b7fc4ffe90a936f0772c36f2b9/dm/.golangci.yml#L18

**Comment by hi-rustin** [comment]
cc: https://github.com/pingcap/tiflow/issues/2162

File: .golangci.yml [hunk 2]
```diff
@@ -4,6 +4,22 @@ linters:
     - unparam
     - revive
     - depguard
+    - misspell
+    - ineffassign
+    - typecheck
+    - varcheck
+    - unused
```
Comment (hi-rustin) [inline .golangci.yml:11]: It seems we have listed some lint that is open by default? But it shouldn't matter.

**Review by hi-rustin — APPROVED** [review]

File: .golangci.yml (hunk 2 above)
Comment (dveeden) [reply .golangci.yml:11]: yes indeed

**Comment by hi-rustin** [comment]
Thanks!

**Pull request merged**